dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/URLConnectionHTTPConduit.java

```diff
@@ -472,7 +475,17 @@ protected boolean usingProxy() {
         protected void setFixedLengthStreamingMode(int i) {
             // [CXF-6227] do not call connection.setFixedLengthStreamingMode(i)
             // to prevent https://bugs.openjdk.java.net/browse/JDK-8044726
+        
+            // Liberty Change for issue #25866-unexpected_EOF_from_server
+            // The bug [CXF-6227] is a performance bug observed for Java 7
+            // In case of a performance drop is observed need to create a walk around
+            boolean isRestMessage = PropertyUtils.isTrue(outMessage.getExchange().get(org.apache.cxf.message.Message.REST_MESSAGE));
+            // Don't call setFixedLengthStreamingMode for JAX-RS
```

Comment:
Are you sure this is still needed for JAXRS?   https://bugs.openjdk.java.net/browse/JDK-8044726 looks like it was resolved in 2015 so perhaps this is no longer a problem.

Reply:
When we don't skip this method call, many JAX-RS tests fail with `java.lang.IllegalStateException: IllegalStateException invoking <URL>: Already connected and the expected response is OK` . I think performance wouldn't be an issue anymore. I agree on that.

Reply:
ok